pkg/cmd/repo/delete/delete.go

```diff
@@ -106,6 +107,17 @@ func deleteRun(opts *DeleteOptions) error {
 
 	err = deleteRepo(httpClient, toDelete)
 	if err != nil {
+		var httpErr api.HTTPError
+		if errors.As(err, &httpErr) {
+			statusCode := httpErr.HTTPError.StatusCode
+			if statusCode == http.StatusMovedPermanently ||
+				statusCode == http.StatusTemporaryRedirect ||
+				statusCode == http.StatusPermanentRedirect {
+				cs := opts.IO.ColorScheme()
+				fmt.Fprintf(opts.IO.ErrOut, "%s Failed to delete repository: %s has changed name or transfered ownership\n", cs.FailureIcon(), fullName)
```

Comment:
Went with this error message as the redirect response from the server does not give the new repository name just an ID.